docs/sigv4_post_sample.py

```diff
@@ -0,0 +1,146 @@
+#   Insights Engine.
+#
+###############################################################################
+#
+# Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# This file is licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+# AWS Version 4 signing example for API Gateway REST API (POST)
+
+# See: http://docs.aws.amazon.com/general/latest/gr/sigv4_signing.html
+# This version makes a POST request and passes request parameters
+# in the body (payload) of the request. Auth information is passed in
+# an Authorization header.
+import sys, os, base64, datetime, hashlib, hmac 
+import requests # pip install requests
+
+# ************* REQUEST VALUES *************
+method = 'POST'
+service = 'execute-api'
+# Host is the base URL for your REST API, where {restapi_id} is the API identifier, {region} is the Region of the API deployment. 
+host = '{restapi_id}.execute-api.{region}.amazonaws.com'
+region = '{region}'
+# Endpoint is the endpoint URL for your REST API resource, where {api_name} is the API name, and {method_name} is the name of the method resource of the API deployment. 
+endpoint = 'https://{restapi_id}.execute-api.us-west-2.amazonaws.com/api/{api_name}/{method_name}'
+# POST requests use a content type header.
+content_type = 'application/json'
+
+# Specify the Amazon S3 location for the input media file:
+s3_bucket = '{s3_bucket}'
+s3_key = '{s3_key}'
+# Request parameters for executing the CasImageWorkflow in its default configuration
+request_parameters =  '{"Name":"CasImageWorkflow", "Input":{"Media":{"Image":{"S3Bucket": "' + s3_bucket + '", "S3Key":"' + s3_key + '"}}}}'
+
+# Key derivation functions. See:
+# http://docs.aws.amazon.com/general/latest/gr/signature-v4-examples.html#signature-v4-examples-python
+def sign(key, msg):
+    return hmac.new(key, msg.encode("utf-8"), hashlib.sha256).digest()
+
+def getSignatureKey(key, date_stamp, regionName, serviceName):
+    kDate = sign(('AWS4' + key).encode('utf-8'), date_stamp)
+    kRegion = sign(kDate, regionName)
+    kService = sign(kRegion, serviceName)
+    kSigning = sign(kService, 'aws4_request')
+    return kSigning
+
+# Read AWS access key from env. variables or configuration file. Best practice is NOT
+# to embed credentials in code.
+access_key = os.environ.get('AWS_ACCESS_KEY_ID')
+secret_key = os.environ.get('AWS_SECRET_ACCESS_KEY')
+if access_key is None or secret_key is None:
+    print('No access key is available.')
+    sys.exit()
+
+# Create a date for headers and the credential string
+t = datetime.datetime.utcnow()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. The recommended way to create an aware datetime object representing a specific timestamp in UTC is by using `now` or `fromtimestamp` methods and passing `tzinfo` to them.

[Learn more](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects)